### ISSUE TITLE
Use `meta.mainProgram` by default in `mkApp`

### ISSUE THREAD

**Issue by yihuang** [numtide/flake-utils#65]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by zimbatm** [comment]
Indeed. mkApp was added before that heuristic was added to nix. TBH I think mkApp should mostly get deprecated.

**Comment by soupglasses** [comment]
Agree on making `mkApp` itself deprecated. It adds little value and is a confusing abstraction.

I would still recommend adding a warning or note with this that explains that `passthru.exePath` is no longer used and `meta.mainProgram` should be used instead. As this would be vastly helpful for anyone who may have set this option in their derivations in the past for `mkApp` support.

**Comment by zimbatm** [comment]
Sounds good, do you want to do a PR for it?

**Comment by soupglasses** [comment]
Opened! :smile:
